src/agent/agent.c

```diff
@@ -1790,13 +1790,62 @@ static int agent_property_get_disconnect_timestamp(
         return sd_bus_message_append(reply, "t", agent->disconnect_timestamp);
 }
 
+/*************************************************************************
+ **** org.eclipse.bluechi.Agent.LogLevel ****************
+ *************************************************************************/
+
+static int agent_property_get_log_level(
+                UNUSED sd_bus *bus,
+                UNUSED const char *path,
+                UNUSED const char *interface,
+                UNUSED const char *property,
+                sd_bus_message *reply,
+                UNUSED void *userdata,
+                UNUSED sd_bus_error *ret_error) {
+        const char *log_level = log_level_to_string(bc_log_get_level());
+        return sd_bus_message_append(reply, "s", log_level);
+}
+
+/*************************************************************************
+ **** org.eclipse.bluechi.Agent.LogTarget ****************
+ *************************************************************************/
+
+static int agent_property_get_log_target(
+                UNUSED sd_bus *bus,
+                UNUSED const char *path,
+                UNUSED const char *interface,
+                UNUSED const char *property,
+                sd_bus_message *reply,
+                UNUSED void *userdata,
+                UNUSED sd_bus_error *ret_error) {
+
+        LogFn logfn = bc_log_get_log_fn();
+
+        const char *log_target = NULL;
+        if (bc_log_to_journald_with_location == logfn) {
+                log_target = BC_LOG_TARGET_JOURNALD;
+        }
+
+        if (bc_log_to_stderr_full_with_location == logfn) {
+                log_target = BC_LOG_TARGET_STDERR;
+        }
+
+        if (bc_log_to_stderr_with_location == logfn) {
+                log_target = BC_LOG_TARGET_STDERR_FULL;
+        }
+
+        return sd_bus_message_append(reply, "s", log_target);
+}
+
 
 static const sd_bus_vtable agent_vtable[] = {
         SD_BUS_VTABLE_START(0),
         SD_BUS_METHOD("CreateProxy", "sss", "", agent_method_create_proxy, 0),
         SD_BUS_METHOD("RemoveProxy", "sss", "", agent_method_remove_proxy, 0),
 
         SD_BUS_PROPERTY("Status", "s", agent_property_get_status, 0, SD_BUS_VTABLE_PROPERTY_EMITS_CHANGE),
+        SD_BUS_PROPERTY("LogLevel", "s", agent_property_get_log_level, 0, SD_BUS_VTABLE_PROPERTY_EMITS_CHANGE),
+        SD_BUS_PROPERTY("LogTarget", "s", agent_property_get_log_target, 0, SD_BUS_VTABLE_PROPERTY_EMITS_CHANGE),
```

Comment:
```suggestion
        SD_BUS_PROPERTY("LogLevel", "s", agent_property_get_log_level, 0, SD_BUS_VTABLE_PROPERTY_EXPLICIT),
        SD_BUS_PROPERTY("LogTarget", "s", agent_property_get_log_target, 0, SD_BUS_VTABLE_PROPERTY_CONST),
```
The log target is constant and for log level we state that its value has to be queried explicitly (not emitting a changed signal or something similar).